cloud/cloud_env.cc

```diff
@@ -15,7 +17,85 @@
 #include "util/filename.h"
 
 namespace rocksdb {
+  
+bool CloudEnvOptions::GetNameFromEnvironment(const char *name, const char *alt, std::string * result) {
+  char *value = getenv(name);
+  if (value == nullptr && alt != nullptr) {
+    value = getenv(alt);
+  }
+  if (value != nullptr) {
+    result->assign(value);
+    return true;
+  } else {
+    return false;
+  }
+}
 
+BucketOptions::BucketOptions(bool is_src) {
+  CloudEnvOptions::GetNameFromEnvironment((is_src) ? "ROCKSDB_CLOUD_SRC_OBJECT_PATH" :"ROCKSDB_CLOUD_DEST_OBJECT_PATH",
+					  "ROCKSDB_CLOUD_OBJECT_PATH", &object_);
+  CloudEnvOptions::GetNameFromEnvironment((is_src) ? "ROCKSDB_CLOUD_SRC_REGION" :"ROCKSDB_CLOUD_DEST_REGION",
+					  "ROCKSDB_CLOUD_REGION", &region_);
+  
+  if (! CloudEnvOptions::GetNameFromEnvironment((is_src) ?
+						"ROCKSDB_CLOUD_SRC_BUCKET_PREFIX" :
+						"ROCKSDB_CLOUD_DEST_BUCKET_PREFIX",
+						"ROCKSDB_CLOUD_BUCKET_PREFIX", &prefix_)) {
+    prefix_ = "rockset.";
+  }
+  std::string bucket;
+  if (CloudEnvOptions::GetNameFromEnvironment((is_src) ? "ROCKSDB_CLOUD_SRC_BUCKET_NAME" :"ROCKSDB_CLOUD_DEST_BUCKET_NAME",
+					      "ROCKSDB_CLOUD_BUCKET_NAME", &bucket)) {
+    SetBucketName(bucket);
+  }
+}
+  
+void BucketOptions::SetBucketName(const std::string & bucket,
+				  const std::string & prefix) {
+  if (!prefix.empty()) {
+    prefix_ = prefix;
+  }
+  
+  bucket_ = bucket;
+  if (bucket_.empty()) {
+    name_.clear();
+  } else {
+    name_ = prefix_ + bucket_;
+  }
+}
+  
+void BucketOptions::Initialize(const std::string & bucket,
```

Comment:
I am not able to understand the relationship between Initialize, SetBucketName and GetNameFromEnvironment. Can you pl put in some code comments on which method is supposed to do what.  It will help us review the code.

Reply:
Added comments as requested